youtube-tui: init at 0.7.0

Ruixi-rebirth opened the pull request:
youtube-tui: init at 0.7.0

###### Description of changes

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

Kranzes commented on pkgs/applications/video/youtube-tui/default.nix:
```diff
@@ -0,0 +1,31 @@
+  };
+  cargoHash = "sha256-qcWuh8qaOQBBebdX3D01k5yXZfifbFC+ZP0d6bJeOr0=";
+  nativeBuildInputs = [ pkg-config python3 ];
+  buildInputs = [ libsixel libxcb pkg-config openssl ];
```
```suggestion
  buildInputs = [ libsixel libxcb openssl ];
```
`pkg-config` should only go in `nativeBuildInputs` since it is only needed at build time.

Ruixi-rebirth replied:
Thank you, I have modified it

Ruixi-rebirth committed:
youtube-tui: init at 0.7.0

youtube-tui: init at 0.7.0

youtube-tui: init at 0.7.0

youtube-tui: init at 0.7.0

youtube-tui: init at 0.7.0

youtube-tui: init at 0.7.0

Kranzes commented on pkgs/applications/video/youtube-tui/default.nix:
```diff
@@ -0,0 +1,31 @@
+  nativeBuildInputs = [ pkg-config python3 ];
+  buildInputs = [ libsixel libxcb openssl ];
+
+  doCheck = false;
```
Please add a comment on why the checks are disabled.

Ruixi-rebirth replied:
aha,no need to set **doCheck=false**

Kranzes commented on pkgs/applications/video/youtube-tui/default.nix:
```diff
@@ -0,0 +1,29 @@
+    sha256 = "sha256-Rz13mGqxfCU+6k2Ejvk+ZoaLirDT6VkmAIhvFdNRvuI=";
+  };
+  cargoHash = "sha256-qcWuh8qaOQBBebdX3D01k5yXZfifbFC+ZP0d6bJeOr0=";
+  nativeBuildInputs = [ pkg-config python3 ];
```
Why is python needed?

Kranzes replied:
Got it, xcb needs it.

Kranzes commented:
I suggest you use this expression instead as it has fixes for darwin too:
```nix
{ lib
, rustPlatform
, fetchFromGitHub
, pkg-config
, openssl
, xorg
, stdenv
, darwin
, python3
, libsixel
}:

rustPlatform.buildRustPackage rec {
  pname = "youtube-tui";
  version = "0.7.0";

  src = fetchFromGitHub {
    owner = "Siriusmart";
    repo = pname;
    rev = "v${version}";
    hash = "sha256-YQj+hmNh8rqP7bKeFDQhZIf79WG7vqg31oReb0jrmg4=";
  };

  cargoHash = "sha256-qcWuh8qaOQBBebdX3D01k5yXZfifbFC+ZP0d6bJeOr0=";

  nativeBuildInputs = [
    pkg-config
    python3
  ];

  buildInputs = [
    openssl
    xorg.libxcb
    libsixel
  ] ++ lib.optionals stdenv.isDarwin [
    darwin.apple_sdk.frameworks.CoreFoundation
    darwin.apple_sdk.frameworks.Security
  ];

  meta = with lib; {
    description = "An aesthetically pleasing YouTube TUI written in Rust";
    homepage = "https://siriusmart.github.io/youtube-tui";
    license = licenses.gpl3Only;
    maintainers = with maintainers; [ Ruixi-rebirth ];
  };
}
```

Kranzes commented:
Can you squash the 3 commits into 1 please? Thanks.

Kranzes reviewed (changes requested):
Also I totally missed it, you need to add the `callPackage` to `pkgs/top-level/all-packages.nix`.

Ruixi-rebirth commented:
> Can you squash the 3 commits into 1 please? Thanks.

Ok, I will rebase after going home

Ruixi-rebirth commented:
> Also I totally missed it, you need to add the `callPackage` to `pkgs/top-level/all-packages.nix`.

everything is fine.

Ruixi-rebirth committed:
maintainers: add Ruixi-rebirth

Kranzes commented on pkgs/applications/video/youtube-tui/default.nix:
```diff
@@ -0,0 +1,50 @@
+  ] ++ lib.optionals stdenv.isDarwin [
+    CoreFoundation
+    Security
+    AppKit
```
is AppKit needed?

Ruixi-rebirth replied:
Yes, the appkit is required, without it the bot checks for youtube.passthru.tests on x86_64-darwin and youtube.passthru.tests on aarch64-darwin will fail to build

Kranzes replied:
I don't have a darwin machine so I have to trust you.
